HW2/P2/parallel_vector.pyx

```diff
@@ -66,27 +66,43 @@ cpdef move_data_serial(np.int32_t[:] counts,
                    counts[src[idx]] -= 1
 
 
+
 cpdef move_data_fine_grained(np.int32_t[:] counts,
                              np.int32_t[:] src,
                              np.int32_t[:] dest,
                              int repeat):
-   cdef:
-       int idx, r
-       omp_lock_t *locks = get_N_locks(counts.shape[0])
+  cdef:
+        int idx, r, small_idx, big_idx
+        omp_lock_t *locks = get_N_locks(counts.shape[0])
 
    ##########
    # Your code here
    # Use parallel.prange() and a lock for each element of counts to parallelize
    # data movement.  Be sure to avoid deadlock, and double-locking.
    ##########
-   with nogil:
-       for r in range(repeat):
-           for idx in range(src.shape[0]):
-               if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
-                   counts[src[idx]] -= 1
+  with nogil:
+    for r in range(repeat):
+        for idx in prange(src.shape[0], num_threads=4, schedule='static'):
+
+          small_idx = min(src[idx], dest[idx])
```

Comment:
The min function was a good idea...I did everything with if statements. Your way is definitely better!